src/components/Navbar/index.jsx

```diff
@@ -33,27 +37,64 @@ const socialIcons = [
 ]
 
 export default function Navbar() {
+  const { toggleTheme, theme } = useTheme()
+  const isChecked = theme === "dark"
+  const { isAuthenticated } = useAuth()
+  const pathname = usePathname()
+
+  const [isOpen, setOpen] = useState(false)
+  const handleOpen = () => {
+    setOpen(!isOpen)
```

Comment:
don't forget, always grab the previousState if you want to change the state based on the previous value. `setOpen(prevState => !prevState)`. This is the recommended way of changing state in react
[Read More Here](https://pt-br.react.dev/reference/react/useState#updating-state-based-on-the-previous-state)